nn_belyi/AI_functions.py

```diff
@@ -0,0 +1,101 @@
+# Functions related to setting up the network.
+
+# INTERESTING: Try to train a network that recognizes the (0,1)-dessign from a (0, inf)-dessign 
+
+###############################################################################################
+# Utilities for creating and fine-tuning neural networks in Keras.
+
+
+import os, sys, scipy.io, scipy.linalg, time, random, pickle
+import warnings
+warnings.simplefilter(action='ignore', category=FutureWarning)
+warnings.simplefilter(action='ignore', category=RuntimeWarning)
+os.environ['TF_CPP_MIN_LOG_LEVEL']='2' #don't display warnings; only errors
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pylab as plt
+
+# Keras imports
+from tensorflow.keras.models import Sequential, load_model
+from tensorflow.keras.layers import Conv2D, MaxPooling2D, Activation, Dropout, Flatten, Dense
+from tensorflow.keras import optimizers
+
+from sklearn.metrics import confusion_matrix
+
+##################################################
+# Local imports
+
+from dessin_data import TrainingOrbit
+
+##################################################
+# Training functions.
+
+def train_belyi(data,
+                PCAk = 23,
```

Comment:
is this used?

Reply:
Hi Michael. Good to hear from you! (I think we met a awhile back, during HMF week.)

The answer is, "eventually yes". This function is intended to take in a bunch of dessin pairs and train a neural network (of some type) to recognize when they are in the same Galois orbit.

Reply:
Yes of course I realize this is WIP ;)

I was commenting here on the plan to use `PCAk` or not

Reply:
Well, I'm not quite sure if it will be used, but I was hoping to keep it around as a reminder to myself. 
If our data is vectorized in a way where a data point is sent to a sparse vector in a high dimensional space, PCA is a pretty good trick of carving down the dimension to get better learning out of the machine learning. Granted, we haven't figured out encoding yet.